Add payment error messages

meandavejustice opened the pull request:
- fixes #1589

![error-dialog](https://user-images.githubusercontent.com/1844554/60986827-4b514e00-a30e-11e9-8dd3-79b51f23ff2d.png)

lmorchard commented on packages/fxa-payments-server/src/lib/errors.ts:
```diff
@@ -1,9 +1,109 @@
 // ref: fxa-auth-server/lib/error.js
-export const AuthServerErrno = {
```
I usually prefer leaving the exports in-line, rather than listing them all at the end. 

That said... there are a lot of intermediate variables here that *aren't* exported. Listing them at the end seems to keep them from getting lost in the rest of the file. So, :+1:

lmorchard commented:
Oh, but it looks like you do need to [set up GPG commit signing](https://micropipes.com/blog/2016/08/31/signing-your-commits-on-github-with-a-gpg-key/) here since we turned on enforcement of that :/

lmorchard commented:
Oh yeah, would probably be nice to add some stories to maybe both the Product & Subscription routes to display some of these new errors. But we could spin that off as a follow-up

meandavejustice commented:
I've added some basic tests for `errors.ts`, updated the commit message to follow our guidelines, and enabled gpg signing but I'm having trouble remember the password for my gpg key 😭. I'm poking around cus I'm pretty sure I wrote it down somewhere this time.

@lmorchard 
> Oh yeah, would probably be nice to add some stories to maybe both the Product & Subscription routes to display & demo some of these new errors. But we could spin that off as a follow-up

I haven't figured out a good way to trigger this error, while developing I was manually adding some characters to the api key to invalidate it. 🤔

lmorchard commented:
> I haven't figured out a good way to trigger this error, while developing I was manually adding some characters to the api key to invalidate it. 

Oh, actually, for storybook purposes, it might be enough to set up some React views with dialogs just to show how the messages render & appear in the layout. Doesn't have to be a part of this PR though
